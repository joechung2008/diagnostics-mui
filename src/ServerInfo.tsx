import {
  Table,
  TableBody,
  TableCell,
  TableHead,
  TableRow,
} from "@mui/material";

const ServerInfo: React.FC<ServerInfoProps> = ({
  deploymentId,
  extensionSync,
  hostname,
  nodeVersions,
  serverId,
  uptime,
}) => {
  const items = [
    {
      name: "Hostname",
      value: hostname,
    },
    {
      name: "Uptime",
      value: uptime,
    },
    {
      name: "Server ID",
      value: serverId,
    },
    {
      name: "Deployment ID",
      value: deploymentId,
    },
    {
      name: "Node Versions",
      value: nodeVersions,
    },
    {
      name: "Extension Sync | Total Sync All Count",
      value: extensionSync.totalSyncAllCount,
    },
  ];

  return (
    <Table aria-label="Server Info">
      <TableHead>
        <TableRow>
          <TableCell>Name</TableCell>
          <TableCell>Value</TableCell>
        </TableRow>
      </TableHead>
      <TableBody>
        {items.map((item) => (
          <TableRow key={item.name}>
            <TableCell>{item.name}</TableCell>
            <TableCell>{item.value}</TableCell>
          </TableRow>
        ))}
      </TableBody>
    </Table>
  );
};

export default ServerInfo;
