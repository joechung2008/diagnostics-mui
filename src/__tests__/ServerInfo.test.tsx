import { render } from "@testing-library/react";
import { describe, expect, it } from "vitest";
import ServerInfo from "../ServerInfo";

describe("ServerInfo", () => {
  it("matches snapshot", () => {
    const props = {
      deploymentId: "deploy-1",
      extensionSync: { totalSyncAllCount: 1 },
      hostname: "localhost",
      nodeVersions: "v18.0.0",
      serverId: "server-1",
      uptime: 100,
    };
    const { container } = render(<ServerInfo {...props} />);
    expect(container).toMatchSnapshot();
  });
});
