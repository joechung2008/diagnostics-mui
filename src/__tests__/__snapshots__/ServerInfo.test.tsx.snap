// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ServerInfo > matches snapshot 1`] = `
<div>
  <table
    aria-label="Server Info"
    class="MuiTable-root css-1xwxv7r-MuiTable-root"
  >
    <thead
      class="MuiTableHead-root css-1a7iywq-MuiTableHead-root"
    >
      <tr
        class="MuiTableRow-root MuiTableRow-head css-1xtozoh-MuiTableRow-root"
      >
        <th
          class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1orzuox-MuiTableCell-root"
          scope="col"
        >
          Name
        </th>
        <th
          class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1orzuox-MuiTableCell-root"
          scope="col"
        >
          Value
        </th>
      </tr>
    </thead>
    <tbody
      class="MuiTableBody-root css-gmh7jj-MuiTableBody-root"
    >
      <tr
        class="MuiTableRow-root css-1xtozoh-MuiTableRow-root"
      >
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1dc80h3-MuiTableCell-root"
        >
          Hostname
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1dc80h3-MuiTableCell-root"
        >
          localhost
        </td>
      </tr>
      <tr
        class="MuiTableRow-root css-1xtozoh-MuiTableRow-root"
      >
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1dc80h3-MuiTableCell-root"
        >
          Uptime
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1dc80h3-MuiTableCell-root"
        >
          100
        </td>
      </tr>
      <tr
        class="MuiTableRow-root css-1xtozoh-MuiTableRow-root"
      >
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1dc80h3-MuiTableCell-root"
        >
          Server ID
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1dc80h3-MuiTableCell-root"
        >
          server-1
        </td>
      </tr>
      <tr
        class="MuiTableRow-root css-1xtozoh-MuiTableRow-root"
      >
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1dc80h3-MuiTableCell-root"
        >
          Deployment ID
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1dc80h3-MuiTableCell-root"
        >
          deploy-1
        </td>
      </tr>
      <tr
        class="MuiTableRow-root css-1xtozoh-MuiTableRow-root"
      >
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1dc80h3-MuiTableCell-root"
        >
          Node Versions
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1dc80h3-MuiTableCell-root"
        >
          v18.0.0
        </td>
      </tr>
      <tr
        class="MuiTableRow-root css-1xtozoh-MuiTableRow-root"
      >
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1dc80h3-MuiTableCell-root"
        >
          Extension Sync | Total Sync All Count
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1dc80h3-MuiTableCell-root"
        >
          1
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
